contracts/contracts/PoolDeposits.sol

```diff
@@ -215,11 +231,24 @@ contract PoolDeposits {
     }
   }
 
+  /*
```

Comment:
I think we can remove this, we basically just need to go through the modifiers and check if  `userHasNoProposal`.